### PR TITLE
Android - added ShortcutBadger support

### PR DESCRIPTION
**Pull request by kfiroo**
A notification object containing `badge` field will set the the `unread notification counter` on the app icon.

A notification without the `badge` field will not affect the counter.

To reset the counter set the `badge` field to 0.

See ShortcutBadger for supported launchers.

### CODE REVIEW

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 1]
```diff
@@ -231,6 +232,21 @@ public void sendNotification(Bundle bundle) {
                 notification.setVibrate(new long[]{0, vibration});
             }
 
+            int badge = -1;
+            if (bundle.containsKey("badge")) {
```
Comment (npomfret) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: Can I suggest you use the 'number' field so its the same as iOS?

Comment (kfiroo) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: According to apple `badge` is the right field to use
https://developer.apple.com/library/ios/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/TheNotificationPayload.html

Comment (npomfret) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: Actually, after a read of the RN [docs](https://facebook.github.io/react-native/docs/pushnotificationios.html) I think it's applicationIconBadgeNumber when included in a notification. 

And there's also a static method `setApplicationIconBadgeNumber`.

Comment (npomfret) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: ...however in this project, if you take a look at [index.js](https://github.com/zo0r/react-native-push-notification/blob/master/index.js#L121) you can see that the `number` field seems to get translated into `applicationIconBadgeNumber` for iOS.

Comment (igrayson) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: If we want iOS parity, `badge` is right key to read. The doc linked is pretty clear, and we can confirm. We use `badge` payload attribute (along side `alert`, `content-available`, and `sound` -- also discussed in the doc)  for our existing iOS-only app.

File: android/build.gradle [hunk 2]
```diff
@@ -27,4 +31,5 @@ dependencies {
     compile 'com.android.support:appcompat-v7:23.1.1'
     compile 'com.facebook.react:react-native:+'
     compile 'com.google.android.gms:play-services-gcm:7.8.0'
+    compile 'me.leolin:ShortcutBadger:1.1.4@aar'
```
Comment (igrayson) [inline android/build.gradle]: Suggest using the latest `1.1.8`

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 3]
```diff
@@ -231,6 +232,21 @@ public void sendNotification(Bundle bundle) {
                 notification.setVibrate(new long[]{0, vibration});
             }
 
+            int badge = -1;
```
Comment (igrayson) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: More clear: `Integer badge;`

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 4]
```diff
@@ -231,6 +232,21 @@ public void sendNotification(Bundle bundle) {
...
+                try {
+                    String badgeAsString = bundle.getString("badge");
+                    if (badgeAsString != null) {
+                        badge = Integer.parseInt(badgeAsString);
+                    }
+                } catch (Exception e) {
+                    badge = -1;
+                }
+            }
+            if (badge > -1) {
```
Comment (igrayson) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: using `Integer`:
```java
if (badge != null && badge >= 0) {

}
```

File: android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java [hunk 5]
```diff
@@ -231,6 +232,21 @@ public void sendNotification(Bundle bundle) {
                 notification.setVibrate(new long[]{0, vibration});
             }
 
+            int badge = -1;
+            if (bundle.containsKey("badge")) {
+                try {
+                    String badgeAsString = bundle.getString("badge");
```
Comment (igrayson) [inline android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java]: This seems wrong; the payload should contain `badge` as a numeric, not a string.

**Comment by igrayson** [comment]
I don't think `RNPushNotificationHelper` is the right place to put this logic. `RNPushNotificationHelper` is not invoked for incoming remote push notifications, while the app is in the foreground.

This logic probably belongs in `RNPushNotificationListenerService`

**Comment by igrayson** [comment]
I've pushed a PR with the approach we've been using: https://github.com/zo0r/react-native-push-notification/pull/181

**Comment by npomfret** [comment]
So will `applicationIconBadgeNumber` work for both iOS and Android?  And will the `number` field remain in use for local notifications, because that is whats being used to create _ applicationIconBadgeNumber_ right now.

**Pull request closed without merging**